mergify_engine/tests/unit/rules/test_filter.py

```diff
@@ -215,6 +217,191 @@ async def test_time_binary() -> None:
     assert await f(FakePR({"foo": time(8, 9)}))
 
 
+@pytest.mark.parametrize(
+    "klass",
+    (
+        date.Day,
+        date.Month,
+        date.Year,
+        date.DayOfWeek,
+    ),
+)
+@freeze_time("2012-01-14")
+async def test_partial_datetime_binary(
+    klass: typing.Type[date.PartialDatetime],
+) -> None:
+    assert "foo>=5" == str(filter.BinaryFilter({">=": ("foo", klass(5))}))
+    assert "foo<=23" == str(filter.BinaryFilter({"<=": ("foo", klass(23))}))
+    assert "foo=3" == str(filter.BinaryFilter({"=": ("foo", klass(3))}))
+
+    f = filter.BinaryFilter({"<=": ("foo", klass(5))})
+    assert await f(FakePR({"foo": klass(2)}))
+    assert await f(FakePR({"foo": klass(5)}))
+    assert not await f(FakePR({"foo": klass(30)}))
+
+    f = filter.BinaryFilter({"=": ("foo", klass(5))})
+    assert await f(FakePR({"foo": klass(5)}))
+    assert not await f(FakePR({"foo": klass(30)}))
+
+    f = filter.BinaryFilter({">": ("foo", klass(5))})
+    assert await f(FakePR({"foo": klass(30)}))
+    assert not await f(FakePR({"foo": klass(2)}))
+    assert not await f(FakePR({"foo": klass(5)}))
+
+
+@freeze_time("2012-01-14T12:15:00")
+async def test_day_near_datetime() -> None:
+    nextday = datetime.datetime(2012, 1, 7, 0, 0, 0, 0, tzinfo=datetime.timezone.utc)
```

Comment:
I'm a bit confused: this says next day is the 7th January but the time is frozen on 14th January? 🤔

Reply:
The test just need the year and month to be frozen, to check override the day. I put 6 by default anyway.